src/index.ts

```diff
@@ -0,0 +1,121 @@
+import { type RemotePinningServiceClient, type Pin, type PinStatus, type PinsRequestidPostRequest, Status } from '@ipfs-shipyard/pinning-service-client'
+import { multiaddr } from '@multiformats/multiaddr'
+import debug from 'debug'
+import pRetry from 'p-retry'
+import { FailedToConnectToDelegates } from './errors.js'
+import type { Helia } from '@helia/interface'
+import type { CID } from 'multiformats/cid'
+
+const log = debug('helia-remote-pinning')
+const logError = log.extend('error')
+const logTrace = log.extend('trace')
+
+export interface HeliaRemotePinningOptions {
+  /**
+   * Control whether requests are aborted or not by manually aborting a signal or using AbortSignal.timeout()
+   */
+  signal?: AbortSignal
+
+  /**
+   * The CID instance to pin. When using Helia, passing around the CID object is preferred over the string.
+   */
+  cid: CID
+}
+
+export interface AddPinArgs extends Omit<Pin, 'cid'>, HeliaRemotePinningOptions {}
+
+export interface ReplacePinArgs extends Omit<PinsRequestidPostRequest, 'pin'>, Omit<Pin, 'cid'>, HeliaRemotePinningOptions {}
+
+export class HeliaRemotePinner {
```

Comment:
I would consider not exporting this class directly, instead just export the factory function that returns an instance of an interface that this class implements.  This gives you flexibility to add extra methods to the class that are not part of the public API.

If you need to invoke those methods during testing, you can define this class in a separate file that you can import from the tests but not add it to the exports map.

Reply:
fixed in https://github.com/ipfs/helia-remote-pinning/pull/2/commits/52d998b4f3da5856f3eaced8474f547a368222fb